src/Versions.jl

```diff
@@ -8,34 +8,42 @@ export VersionBound, VersionRange, VersionSpec, semver_spec, isjoinable
 # VersionBound #
 ################
 struct VersionBound
-    t::NTuple{3,UInt32}
+    t::NTuple{4,UInt32}
     n::Int
     function VersionBound(tin::NTuple{n,Integer}) where n
-        n <= 3 || throw(ArgumentError("VersionBound: you can only specify major, minor and patch versions"))
-        n == 0 && return new((0,           0,      0), n)
-        n == 1 && return new((tin[1],      0,      0), n)
-        n == 2 && return new((tin[1], tin[2],      0), n)
-        n == 3 && return new((tin[1], tin[2], tin[3]), n)
+        n <= 4 || throw(ArgumentError("VersionBound: you can only specify major, minor, patch versions, and a build number"))
+        n == 0 && return new((0,           0,      0,      0), n)
+        n == 1 && return new((tin[1],      0,      0,      0), n)
+        n == 2 && return new((tin[1], tin[2],      0,      0), n)
+        n == 3 && return new((tin[1], tin[2], tin[3],      0), n)
+        n == 4 && return new((tin[1], tin[2], tin[3], tin[4]), n)
         error("invalid $n")
     end
 end
 VersionBound(t::Integer...) = VersionBound(t)
-VersionBound(v::VersionNumber) = VersionBound(v.major, v.minor, v.patch)
+function VersionBound(v::VersionNumber)
+    isempty(v.build) && return VersionBound(v.major, v.minor, v.patch)
+    length(v.build) > 1 && @debug "Pkg does not recognize build numbers beyond the first number. These will be ignored" v
+    !(v.build[1] isa Integer) && @debug "Pkg does not recognize string build fields. This will be ignored" v
```

Comment:
You need to return early here. If you try to call the next line, the `VersionBound` constructor will throw an error when you try to convert a string to an `UInt32`.